src/components/molecules/Button/Button.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, PropsWithChildren, SVGProps, FunctionComponent } from 'react'
+import BaseButton, {
+  BaseButtonProps,
+} from 'components/atoms/BaseButton/BaseButton'
+import classNames from 'classnames'
+import classes from './styles.module.scss'
+
+export enum AppearanceTypes {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  Text = 'text',
+}
+
+export enum SizeTypes {
+  M = 'm',
+  S = 's',
+}
+
+export interface ButtonProps extends BaseButtonProps {
+  appearance?: AppearanceTypes
+  size?: SizeTypes
+  icon?: FunctionComponent<SVGProps<SVGSVGElement>>
+  ariaLabel?: string
+  hidden?: boolean
+  disabled?: boolean
+  className?: string
+}
+
+export type IconProps = {
+  icon?: FunctionComponent<SVGProps<SVGSVGElement>>
+  className?: string
+  ariaLabel?: string
+}
+
+const Icon: FC<IconProps> = ({ icon: IconComponent, className, ariaLabel }) => {
+  const ariaLabelToUse = ariaLabel
+
+  if (!IconComponent) return null
+  return (
+    <IconComponent
+      className={(classes.icon, className)}
+      aria-label={ariaLabelToUse}
+    />
+  )
+}
+
+const Button: FC<PropsWithChildren<ButtonProps>> = ({
+  appearance = 'primary',
+  size = 'm',
+  icon,
+  ariaLabel,
+  hidden,
+  disabled,
+  className,
+  children,
+  ...rest
+}) => {
+  if (hidden) return null
+  return (
+    <BaseButton
+      className={classNames(
+        classes.btn,
+        classes[`${appearance}`],
+        classes[`${size}`],
+        classes.baseButton,
+        className
+      )}
+      disabled={disabled}
+      {...rest}
+    >
+      <span className={classes.buttonText}>{children}</span>
+      <Icon icon={icon} className={className} ariaLabel={ariaLabel} />
```

Comment:
**[blocker]**

Not sure if there is a case, where we need to pass the same className to both the outer container (BaseButton) and the icon inside it. However even if there is, it should be 2 separate props. Currently it is nearly impossible to use the className prop in any meaningful way. For example the classic usage for it would be to position the component. In this case we would be positioning both the button component and the icon inside the button at the same time, which would break the button.

The intuitive behaviour for the "className" prop is that it is always passed to the outer most container of the component. That way it will be used the same way as the className prop of native html elements.